test/NuGet.Core.FuncTests/NuGet.Commands.FuncTest/RestoreCommand_AlgorithmEquivalencyTests.cs

```diff
@@ -52,6 +52,56 @@ public async Task RestoreCommand_HigherLevelSuppressionsWin_VerifiesEquivalency(
             result.LockFile.Targets[0].Libraries[2].Name.Should().Be("X");
         }
 
+        // LightGbm (project) -> Microsoft.ML.LightGbm -> LightGBM 2.0.0 (package)
+        // This should result in an error with the cycle
+        [Fact]
+        public async Task RestoreCommand_WithCycleContainingProject_FailsAndLogsMessage()
+        {
+            // Arrange
+            using var pathContext = new SimpleTestPathContext();
+            await SimpleTestPackageUtility.CreateFolderFeedV3Async(
+                pathContext.PackageSource,
+                PackageSaveMode.Defaultv3,
+                new SimpleTestPackageContext("Microsoft.ML.LightGbm", "4.0.1")
+                {
+                    Dependencies =
+                    [
+                        new SimpleTestPackageContext("LightGBM", "3.3.5"),
+                    ]
+                });
+
+            PackageSpec projectSpec = ProjectTestHelpers.GetPackageSpecWithProjectNameAndSpec(
+                "LightGbm",
+                pathContext.SolutionRoot,
+                @"
+                {
+                    ""frameworks"": {
+                    ""net472"": {
+                        ""dependencies"": {
+                                ""Microsoft.ML.LightGbm"": ""4.0.1""
+                        }
+                    }
+                    }
+                }");
+
+            // Act & Assert
```

Comment:
I'm not seeing an assertion here. Should this just be Act?

Reply:
Yeah its not obvious that `ValidateRestoreAlgorithmEquivalency` runs assertions.  We could consider renaming it to something like `RestoreAndAssertAlgorithmEquivalency`.